provider/server/server.go

```diff
@@ -68,9 +72,24 @@ func setOptionsDefaults(o *Options) {
 	}
 }
 
+func convertGuestAgentFromString(guestAgent string) (api.GuestAgent, error) {
+	availableGuestAgents := api.GuestAgentAvailable()
```

Comment:
i had switch as first solution but you have to handle options in 2 packages (api and here). I don't like it. It can be harder handle it in future and this function is calling just one time during start, so i don't see big benefits in performance improvement with switch.

Reply:
GuestAgentAvailable was moved